util/http.go

```diff
@@ -58,44 +58,21 @@ func GetXMatrixAuth(headers []string) ([]XMatrixAuth, error) {
 
 		paramCsv := h[len("X-Matrix "):]
 		params := make(map[string]string)
-		isKey := true
-		keyName := ""
-		keyValue := ""
-		escape := false
-		for _, c := range paramCsv {
-			if c == ',' && isKey {
-				params[strings.TrimSpace(strings.ToLower(keyName))] = keyValue
-				keyName = ""
-				keyValue = ""
-				continue
-			}
-			if c == '=' {
-				isKey = false
-				continue
+
+		pairs := strings.Split(paramCsv, ",")
```

Comment:
Won't this break if `,` appears inside a quoted value?

Reply:
probably, but also `,` is illegal inside a value, regardless of quotations.

Reply:
(there's also no possible value which contains `,`)